src/mds/MDCache.cc

```diff
@@ -12035,6 +12035,20 @@ void MDCache::show_subtrees(int dbl)
     }
   }
 
+  if (subtrees_seen.size() > SUBTREES_COUNT_THRESHOLD && depth >
```

Comment:
I think we can be a little smarter and move these checks up to the while loop above so we don't traverse all the subtrees only to do an early return here.